lib/QueryParser.js

```diff
@@ -499,7 +499,11 @@ function getRawSchemaFieldPath(fieldPath, rawSchemas = {}) {
     return paths[fieldPath]
   }
 
-  const patternProperties = Object.keys(rawSchemas.patternProperties || {})
+  // we do this sort to match the most strictest regex
```

Comment:
I suppose this means from the longest to the shorted. Could you elaborate on the reason why this is necessary?

Reply:
Yep, exactly. This was needed because in some cases we add shortes regex matching the pattern and returning the element.
In particular, we could have something like `environments.0.dashboards` matching a RegExp like `"environments\\.\\d+\\..+$"` even if the RegExp `environments\\.\\d+\\.dashboards$` was in the `patternProperties`

Reply:
I described more in the comment as decided